src/hooks/useArbTokenBridge.ts

```diff
@@ -242,14 +241,14 @@ export const useArbTokenBridge = (
   }
 
   const depositToken = async (erc20Address: string, amount: string) => {
-    const _tokenData = await bridge.getAndUpdateL1TokenData(erc20Address)
-    if (!(_tokenData && _tokenData.ERC20)) {
-      throw new Error('Token data not found')
-    }
-    const tokenData = _tokenData.ERC20
+    const tokenData = await bridge.l1Bridge.getL1TokenData(erc20Address)
+
     const amountParsed = await utils.parseUnits(amount, tokenData.decimals)
```

Comment:
I think the UI should give the already parsed amount to the sdk
The UI can then use the token list and we save on these network queries

Reply:
how about UI passes in amount and decimals (we still want to display amount in the txn list)

Reply:
passing amount and decimals seems reasonable but i tend to prefer our internal representations to always be the formatted value.
We can instead keep the formatted value and decimals in the txn list, and adjust that as needed

Reply:
it ooo it'd also have to pass in symbol (which is fine)

Reply:
"t i tend to prefer our internal representations to always be the formatted value." only thing is if we pass in the formatted value as param, we'd have to format it in the UI and then unformat it in the sdk, which we can do, but is a bit awk